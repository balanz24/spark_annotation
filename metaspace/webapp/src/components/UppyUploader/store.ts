// based on https://uppy.io/docs/stores/#Implementing-Stores

import { ref } from '@vue/composition-api'
import { State } from '@uppy/core'

type Listener = (prevState: State, nextState: State, patch: object) => void

export default () => {
  const state = ref<State>({})
  const listeners = new Set<Listener>()

  return {
    getState: () => state.value,
    setState: (patch: object) => {
      const prevState = state.value
      const nextState = { ...prevState, ...patch }

      state.value = nextState

      listeners.forEach(listener => {
        listener(prevState, nextState, patch)
      })
    },
    subscribe: (listener: Listener) => {
      listeners.add(listener)
      return () => listeners.delete(listener)
    },
  }
}
