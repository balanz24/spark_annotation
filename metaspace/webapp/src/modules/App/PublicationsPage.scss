$className: 'sm-publications-page';
$header-height: 64px;

.#{$className} {
  @apply flex flex-col flex-wrap items-center;
  min-height: calc(100vh - #{$header-height});

  p {
    @apply my-0 max-w-measure-4;
  }

  h1 {
    @apply text-4xl leading-12 font-display;
  }

  .sm-content-page.sm-main-content { /* .sm-main-content is added to page routes */
    @apply p-18 max-w-4xl box-border mx-auto leading-6 text-base;
  }

  .sm-content-page > * {
    @apply my-0;
  }

  .sm-content-page > *:not(h1):not(h2) + *,
  .sm-content-page > ul li + li {
    @apply mt-6;
  }

  .sm-content-page > h1 {
    @apply text-4xl leading-12 font-display;
  }

  .sm-content-page > h2 {
    @apply text-2xl leading-6 font-display pb-3 pt-9; /* top padding is for anchor links */
    scroll-margin-top: theme('spacing.12');
  }

  .sm-content-page h3 {
    @apply font-medium;
  }

  .sm-content-page ul {
    @apply p-0 my-0;
  }

  .sm-content-page  ol.sm-ordered-list {
    list-style: none;
    counter-reset: primary-nums;
  }

  .sm-content-page  ol.sm-ordered-list > li {
    @apply mt-6 relative;
  }

  .sm-content-page ol.sm-ordered-list > li::before {
    @apply absolute w-6 h-6 rounded-full bg-blue-100 text-blue-700
    font-bold text-sm flex items-baseline justify-center;

    counter-increment: primary-nums;
    content: counter(primary-nums);
    left: calc(-1 * theme('spacing.9'))
  }

  .sm-content-page p {
    @apply my-0 max-w-measure-4;
  }

  .sm-content-page b {
    @apply font-medium;
  }

  .sm-content-page table {
    @apply border-collapse;
  }

  .sm-content-page td {
    @apply p-0 align-top;
  }

  .sm-content-page code {
    @apply bg-gray-100 px-1 rounded-sm leading-normal;
    display: inline-block;
    font-family: monospace;
  }
}
