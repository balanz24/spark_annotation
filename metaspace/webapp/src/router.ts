/* eslint-disable vue/max-len */
import Vue, { AsyncComponent } from 'vue'
import VueRouter, { RawLocation } from 'vue-router'
import AboutPage from './modules/App/AboutPage'
import DatasetsPage from './modules/Datasets/DatasetsPage.vue'
import { DialogPage, ResetPasswordPage } from './modules/Account'
import { redirectAfterSignIn } from './modules/Account/signInReturnUrl'
import { updateDBParam } from './modules/Filters/url'
import NotFoundPage from './modules/App/NotFoundPage.vue'

Vue.use(VueRouter)

const asyncPagesFreelyTyped = {
  AnnotationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "AnnotationsPage" */ './modules/Annotations/AnnotationsPage.vue'),
  DatasetSummary: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetSummary" */ './modules/Datasets/summary/DatasetSummary.vue'),
  MetadataEditPage: () => import(/* webpackPrefetch: true, webpackChunkName: "MetadataEditPage" */ './modules/MetadataEditor/MetadataEditPage.vue'),
  ImageAlignmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "ImageAlignmentPage" */ './modules/ImageAlignment/ImageAlignmentPage.vue'),
  UploadPage: () => import(/* webpackPrefetch: true, webpackChunkName: "UploadPage" */ './modules/MetadataEditor/UploadPage.vue'),
  DatasetOverviewPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/overview/DatasetOverviewPage'),
  DatasetComparisonPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/comparison/DatasetComparisonPage'),
  DatasetBrowserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/imzml/DatasetBrowserPage'),
  DatasetEnrichmentPage: () => import(/* webpackPrefetch: true, webpackChunkName: "DatasetOverviewPage" */ './modules/Datasets/enrichment/DatasetEnrichmentPage'),
  SpottingProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "SpottingProjectPage" */ './modules/SpottingProject/DashboardPage'),

  // These pages are relatively small as they don't have any big 3rd party dependencies, so pack them together
  DatasetTable: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Datasets/list/DatasetTable.vue'),
  HelpPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/HelpPage.vue'),
  EditUserPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/UserProfile/EditUserPage.vue'),
  CreateGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/CreateGroupPage.vue'),
  ProjectsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Project/ProjectsListPage.vue'),
  SystemHealthPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Admin/SystemHealthPage.vue'),
  GroupsListPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/Group/GroupsListPage'),
  PrivacyPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PrivacyPage.vue'),
  TermsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/TermsPage.vue'),
  PublicationsPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle1" */ './modules/App/PublicationsPage'),

  // These pages use sanitizeHtml, which is big
  ViewGroupPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Group/ViewGroupPage.vue'),
  ViewProjectPage: () => import(/* webpackPrefetch: true, webpackChunkName: "Bundle2" */ './modules/Project/ViewProjectPage.vue'),

  // Separate bundle for design docs
  DesignTOC: () => import(/* webpackChunkName: "DesignBundle" */ './design/TOCPage.vue'),
  DesignStyleGuide: () => import(/* webpackChunkName: "DesignBundle" */ './design/StyleGuidePage.vue'),
  DesignIcons: () => import(/* webpackChunkName: "DesignBundle" */ './design/IconsPage.vue'),
  DesignComponents: () => import(/* webpackChunkName: "DesignBundle" */ './design/ComponentsPage.vue'),
  DesignForms: () => import(/* webpackChunkName: "DesignBundle" */ './design/FormsPage.vue'),
}
const asyncPages = asyncPagesFreelyTyped as Record<keyof typeof asyncPagesFreelyTyped, AsyncComponent>

const convertLegacyUrls = () => {
  const { pathname, hash, search } = window.location
  if (pathname === '/' && hash && hash.startsWith('#/')) {
    history.replaceState(undefined, undefined as any, hash.slice(1))
  }
  if (pathname === '/annotations') {
    const updatedQueryString = updateDBParam(search.slice(1))
    if (updatedQueryString !== null) {
      history.replaceState(undefined, undefined as any, `${pathname}?${updatedQueryString}`)
    }
  }
}
convertLegacyUrls()

const router = new VueRouter({
  mode: 'history',
  routes: [
    { path: '/(about)?', component: AboutPage, meta: { footer: true, headerClass: 'bg-primary' } },
    { path: '/annotations', name: 'annotations', component: asyncPages.AnnotationsPage },
    {
      path: '/datasets',
      component: DatasetsPage,
      children: [
        { path: '', component: asyncPages.DatasetTable },
        { path: 'summary', component: asyncPages.DatasetSummary },
      ],
    },
    {
      path: '/datasets/:dataset_id/comparison',
      name: 'datasets-comparison',
      component: asyncPages.DatasetComparisonPage,
    },
    { path: '/datasets/edit/:dataset_id', name: 'edit-metadata', component: asyncPages.MetadataEditPage },
    { path: '/datasets/:dataset_id/add-optical-image', name: 'add-optical-image', component: asyncPages.ImageAlignmentPage },
    { path: '/dataset/:dataset_id', name: 'dataset-overview', component: asyncPages.DatasetOverviewPage },
    { path: '/dataset/:dataset_id/annotations', name: 'dataset-annotations', component: asyncPages.AnnotationsPage },
    { path: '/dataset/:dataset_id/browser', name: 'dataset-browser', component: asyncPages.DatasetBrowserPage },
    { path: '/dataset/:dataset_id/enrichment', name: 'dataset-enrichment', component: asyncPages.DatasetEnrichmentPage },
    { path: '/upload', component: asyncPages.UploadPage },
    { path: '/help', component: asyncPages.HelpPage, meta: { footer: true } },
    { path: '/user/me', component: asyncPages.EditUserPage },

    { path: '/admin/health', component: asyncPages.SystemHealthPage },

    { path: '/account/sign-in', component: DialogPage, props: { dialog: 'signIn' } },
    { path: '/account/sign-in-success', redirect: redirectAfterSignIn },
    { path: '/account/create-account', component: DialogPage, props: { dialog: 'createAccount' } },
    { path: '/account/forgot-password', component: DialogPage, props: { dialog: 'forgotPassword' } },
    { path: '/account/reset-password', component: ResetPasswordPage },

    { path: '/groups', component: asyncPages.GroupsListPage },
    { path: '/group/create', component: asyncPages.CreateGroupPage },
    { path: '/group/:groupIdOrSlug', name: 'group', component: asyncPages.ViewGroupPage },

    { path: '/project/:projectIdOrSlug', name: 'project', component: asyncPages.ViewProjectPage },
    { // Legacy URL sent in "request access" emails up until Feb 2019
      path: '/project/:projectIdOrSlug/manage',
      redirect: { path: '/project/:projectIdOrSlug', query: { tab: 'members' } } as RawLocation,
    },
    { path: '/projects', component: asyncPages.ProjectsListPage },

    { path: '/terms', component: asyncPages.TermsPage, meta: { footer: true } },
    { path: '/privacy', component: asyncPages.PrivacyPage, meta: { footer: true } },
    { path: '/publications', component: asyncPages.PublicationsPage, meta: { footer: true } },

    { path: '/design', component: asyncPages.DesignTOC, meta: { footer: true, flex: true } },
    { path: '/design/styleguide', component: asyncPages.DesignStyleGuide, meta: { footer: true, flex: true } },
    { path: '/design/icons', component: asyncPages.DesignIcons, meta: { footer: true, flex: true } },
    { path: '/design/components', component: asyncPages.DesignComponents, meta: { footer: true, flex: true } },
    { path: '/design/forms', component: asyncPages.DesignForms, meta: { footer: true, flex: true } },

    { path: '/detectability', name: 'detectability', component: asyncPages.SpottingProjectPage },

    { path: '*', component: NotFoundPage, meta: { footer: true, flex: true } },
  ],
  scrollBehavior(to, from, savedPosition) {
    if (to.hash) {
      return { selector: to.hash, offset: { x: 0, y: 64 } } // offset header height
    }
    if (savedPosition) {
      return savedPosition
    }
    if (to.path !== from.path) {
      return { x: 0, y: 0 }
    }
  },
})

const pageLoadedAt = Date.now()
router.beforeEach((to, from, next) => {
  // If user has had the window open for a long time, it's likely they're using old code. This can cause issues
  // such as incorrect API usage, or causing error reports for bugs that have already been fixed.
  // Force the browser to reload the page, preferring to do so during page transitions to minimize impact.
  const daysSincePageLoad = (Date.now() - pageLoadedAt) / 1000 / 60 / 60 / 24
  if (to.path !== from.path && daysSincePageLoad > 1.75 || daysSincePageLoad > 7) {
    window.location.assign(to.fullPath)
    next(false)
  } else {
    next()
  }
})

const { href } = router.resolve({ name: 'project', params: { projectIdOrSlug: 'REMOVE' } }, undefined, true)
export const PROJECT_URL_PREFIX = location.origin + href.replace('REMOVE', '')

export default router
