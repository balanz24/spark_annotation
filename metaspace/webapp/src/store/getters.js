import { mzFilterPrecision } from '../lib/util';
import { decodeParams, decodeSettings, getLevel } from '../modules/Filters';
import config from '../lib/config';

/** For filters where empty string is a valid client-side value that must be converted to empty string for the API */
const noneToEmptyString = s => s === 'none' ? '' : s;

export default {
  filterLevel(state) {
    return getLevel(state.route.path);
  },

  filter(state) {
    return decodeParams(state.route, state.filterLists);
  },

  currentUser(state) {
    return state.currentUser;
  },

  settings(state) {
    return decodeSettings(state.route);
  },

  ftsQuery(state, getters) {
    return getters.filter.simpleQuery;
  },

  gqlAnnotationFilter(state, getters) {
    const filter = getters.filter;
    const colocalizationAlgo = getters.settings.annotationView.colocalizationAlgo;

    const f = {
      compoundQuery: filter.compoundName,
      termId: filter.term ? parseInt(filter.term, 10) : undefined,
      chemMod: noneToEmptyString(filter.chemMod),
      neutralLoss: noneToEmptyString(filter.neutralLoss),
      adduct: filter.adduct,
      annotationId: filter.annotationIds ? filter.annotationIds.join('|') : undefined,
      fdrLevel: filter.fdrLevel,
      pValue: filter.pValue,
      colocalizedWith: filter.colocalizedWith,
      // Only include colocalizationAlgo if there is another filter that uses it. Otherwise the annotations list
      // refreshes unnecessarily when changing algorithm.
      colocalizationAlgo: filter.colocalizedWith || filter.colocalizationSamples ? colocalizationAlgo : null,
      colocalizationSamples: filter.colocalizationSamples,
      offSample: filter.offSample == null ? undefined : !!filter.offSample,
    };

    if (!config.features.neutral_losses && !config.features.advanced_ds_config) {
      f.hasNeutralLoss = false;
    }

    if (!config.features.chem_mods && !config.features.advanced_ds_config) {
      f.hasChemMod = false;
    }

    if (!config.features.all_adducts) {
      f.hasHiddenAdduct = false;
    }

    if (filter.minMSM)
      f.msmScoreFilter = { min: filter.minMSM, max: 1.0 };

    if (filter.mz) {
      const mz = parseFloat(filter.mz),
        deltamz = parseFloat(mzFilterPrecision(filter.mz));
      f.mzFilter = {
        min: (mz - deltamz),
        max: (mz + deltamz)
      };
    }

    f.databaseId = filter.database

    return f;
  },

  gqlDatasetFilter(state, getters) {
    const filter = getters.filter;
    const { group, project, submitter, datasetIds, polarity,
      organism, organismPart, condition, growthConditions,
      ionisationSource, analyzerType, maldiMatrix, metadataType,
      compoundName, datasetOwner, opticalImage } = filter;
    const level = getters.filterLevel;
    const isLogged = state.currentUser && state.currentUser.id
    const hasAnnotationMatching = level === 'dataset' && compoundName ? { compoundQuery: compoundName } : undefined;
    return {
      group: datasetOwner && datasetOwner !== 'my-datasets' && isLogged ? datasetOwner : group,
      project: project,
      submitter: datasetOwner === 'my-datasets' && isLogged ? state.currentUser.id : submitter,

      // temporary workaround because of array-related bugs in apollo-client
      ids: datasetIds ? datasetIds.join("|") : null,

      organism,
      opticalImage,
      organismPart,
      condition,
      growthConditions,
      ionisationSource,
      maldiMatrix,
      analyzerType,
      polarity: polarity ? polarity.toUpperCase() : null,
      metadataType,
      hasAnnotationMatching,
    }
  },

  gqlColocalizationFilter(state, getters) {
    const { datasetIds, colocalizedWith, database, fdrLevel } = getters.filter;
    const colocalizationAlgo = getters.settings.annotationView.colocalizationAlgo;
    if (datasetIds && !datasetIds.includes('|') && colocalizedWith != null && database != null && fdrLevel != null) {
      return { colocalizedWith, colocalizationAlgo, databaseId: database, fdrLevel };
    } else {
      return null;
    }
  },
}
