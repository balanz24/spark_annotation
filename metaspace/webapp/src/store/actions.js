import apolloClient from '../api/graphqlClient';
import {fetchOptionListsQuery} from '../api/metadata';
import {decodeParams} from '../modules/Filters';
import {computed} from "@vue/composition-api";
import store from "../store/index";
export default {

  async initFilterLists(context) {
    if(context.state.filterListsLoading || context.state.filterLists != null)
      return;

    context.commit('setFilterListsLoading');

    const response = await apolloClient.query({
      query: fetchOptionListsQuery,
      fetchPolicy: 'cache-first',
    });

    // set annotationIds default according to values passed from snapshot to the store state
    context.commit('setFilterLists', {...response.data,
      annotationIds: computed(() => store.state.snapshotAnnotationIds)} );

    // Refresh the current filter so that computed defaults that depend on `filterLists` are applied
    const filter = decodeParams(context.state.route, context.state.filterLists)
    context.commit('updateFilter', filter)
  },

};
