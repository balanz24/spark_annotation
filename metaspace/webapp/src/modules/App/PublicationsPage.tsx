import { computed, defineComponent } from '@vue/composition-api'
import { CiteMetaspace } from './CiteMetaspace'
import { useQuery } from '@vue/apollo-composable'
import { countPublicationsQuery, countReviewsQuery } from '../../api/group'
import PublicationItem from './PublicationItem'
import StatefulIcon from '../../components/StatefulIcon.vue'
import { ExternalWindowSvg } from '../../design/refactoringUIIcons'
import './PublicationsPage.scss'

interface Props {
  className: string
}

const PublicationsPage = defineComponent<Props>({
  name: 'PublicationsPage',
  props: {
    className: {
      type: String,
      default: 'publications-page',
    },
  },
  setup(props, ctx) {
    const {
      result: publicationCountResult,
    } = useQuery<{countPublications: any}>(countPublicationsQuery)
    const publicationCount = computed(() => publicationCountResult.value != null
      ? publicationCountResult.value.countPublications : null)
    const {
      result: reviewsCountResult,
    } = useQuery<{countReviews: any}>(countReviewsQuery)
    const reviewCount = computed(() => reviewsCountResult.value != null
      ? reviewsCountResult.value.countReviews : null)

    return () => {
      return (
        <div class='sm-publications-page'>
          <div class="sm-content-page sm-main-content">
            <h1>Relevant Publications</h1>
            <h2 id="ours">
              Our publications presenting METASPACE or its methods (5)
            </h2>
            <ul>
              <PublicationItem
                title="Spatial Metabolomics and Imaging Mass Spectrometry in the Age of Artificial Intelligence"
                authors="Alexandrov T"
                publisher="Annu. Rev. Biomed. Data Sci."
                year="2020"
                link="https://doi.org/10.1146/annurev-biodatasci-011420-031537"
              />
              <PublicationItem
                title="OffsampleAI: artificial intelligence approach to recognize off-sample mass spectrometry images"
                authors="Ovchinnikova K, Kovalev V, Stuart L, Alexandrov T"
                publisher="BMC Bioinformatics"
                year="2020"
                link="https://doi.org/10.1186/s12859-020-3425-x"
              />
              <PublicationItem
                title="ColocML: Machine learning quantifies co-localization between mass spectrometry images"
                authors="Ovchinnikova K, Stuart L, Rakhlin A, Nikolenko S, Alexandrov T"
                publisher="Bioinformatics"
                year="2020"
                link="https://doi.org/10.1093/bioinformatics/btaa085"
              />
              <PublicationItem
                title="METASPACE: A community-populated knowledge base of spatial metabolomes in health and disease"
                authors="Alexandrov T, Ovchnnikova K, Palmer A, Kovalev V, et al."
                publisher="BioRxiv"
                year="2019"
                link="https://www.biorxiv.org/content/10.1101/539478v1.abstract"
              />
              <PublicationItem
                title="FDR-controlled metabolite annotation for high-resolution imaging mass spectrometry"
                authors={'Palmer A, Phapale P, Chernyavsky I, Lavigne R, Fay D, Tarasov A, Kovalev V, Fuchser J, '
                  + 'Nikolenko S, Pineau C, Becker M, Alexandrov T'}
                publisher="Nat. Methods"
                year="2017"
                link="https://doi.org/10.1038/nmeth.4072"
              />
            </ul>

            <h2 id="others">
              Other publications
            </h2>
            <a
              class="publication-link flex items-center"
              href={'https://scholar.google.com/scholar?q=%22METASPACE%22+AND+%28%22imaging+'
                + 'mass+spectrometry%22+OR+%22mass+spectrometry+imaging%22%29'}
              target="_blank"
            >
              Research articles using METASPACE ({(publicationCount.value || 332).toLocaleString()})
              <StatefulIcon class="h-4 w-4 pointer-events-none">
                <ExternalWindowSvg />
              </StatefulIcon>
            </a>
            <a
              class="publication-link flex items-center mt-1"
              href={'https://scholar.google.com/scholar?q=%22METASPACE%22+AND+%28%22imaging+'
                + 'mass+spectrometry%22+OR+%22mass+spectrometry+imaging%22%29++AND+%22Review+article%22&btnG='}
              target="_blank"
            >
              Reviews ({(reviewCount.value || 34).toLocaleString()})
              <StatefulIcon class="h-4 w-4 pointer-events-none">
                <ExternalWindowSvg />
              </StatefulIcon>
            </a>
            <CiteMetaspace/>
          </div>
        </div>
      )
    }
  },
})

export default PublicationsPage
