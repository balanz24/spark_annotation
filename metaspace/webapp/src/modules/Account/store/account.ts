import { Module } from 'vuex'
import { DialogType } from '../dialogs'
import router from '../../../router'
import { RawLocation } from 'vue-router/types/router'

export interface AccountState {
  dialog: DialogType | null;
  dialogCloseRedirect: RawLocation | null;
  loginSuccessRedirect: RawLocation | null;
}

interface ShowDialogOptions {
  dialog: DialogType;
  dialogCloseRedirect?: RawLocation;
  loginSuccessRedirect?: RawLocation;
}

interface HideDialogOptions {
  dialog: DialogType;
  isLoginSuccess?: boolean;
}

const account: Module<AccountState, any> = {
  namespaced: true,
  state: {
    dialog: null,
    dialogCloseRedirect: null,
    loginSuccessRedirect: null,
  },
  mutations: {
    showDialog(state, payload: DialogType | ShowDialogOptions) {
      const options: ShowDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      state.dialog = options.dialog
      if (options.dialogCloseRedirect != null) {
        state.dialogCloseRedirect = options.dialogCloseRedirect
      }
      if (options.loginSuccessRedirect != null) {
        state.loginSuccessRedirect = options.loginSuccessRedirect
      }
    },
    hideDialog(state, payload: DialogType | HideDialogOptions) {
      const options: HideDialogOptions = typeof payload === 'string' ? { dialog: payload } : payload

      if (state.dialog === options.dialog) {
        state.dialog = null
        if (options.isLoginSuccess && state.loginSuccessRedirect != null) {
          router.push(state.loginSuccessRedirect)
        } else if (state.dialogCloseRedirect != null) {
          router.push(state.dialogCloseRedirect)
        }

        state.dialogCloseRedirect = null
        state.loginSuccessRedirect = null
      }
    },
  },
}

export default account
