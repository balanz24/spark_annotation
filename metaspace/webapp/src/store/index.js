import Vue from 'vue';
import Vuex from 'vuex';
Vue.use(Vuex);

import getters from './getters.js';
import mutations from './mutations.js';
import actions from './actions.js';
import {accountModule} from '../modules/Account';

const store = new Vuex.Store({
  state: {
    // names of currently shown filters
    orderedActiveFilters: [],

    filterLists: null,
    filterListsLoading: false,

    // currently selected annotation
    annotation: undefined,

    // currently selected normalization matrix
    normalization: undefined,

    // roi settings
    roiInfo: {visible: false},

    // is annotation table loading?
    tableIsLoading: true,

    lastUsedFilters: {},

    currentTour: null,

    currentUser: {},

    // ion image global viewer settings
    channels: [],
    mode: 'SINGLE'
  },

  getters,
  mutations,
  actions,
  modules: {
    account: accountModule
  }
})

export default store;
