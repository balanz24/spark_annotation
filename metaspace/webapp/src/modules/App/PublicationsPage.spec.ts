import { mount, shallowMount } from '@vue/test-utils'
import PublicationsPage from './PublicationsPage'
import router from '../../router'
import store from '../../store/index'
import Vue from 'vue'
import Vuex from 'vuex'
import { sync } from 'vuex-router-sync'
import { initMockGraphqlClient, apolloProvider } from '../../../tests/utils/mockGraphqlClient'

describe('PublicationsPage', () => {
  const graphqlWithData = () => {
    initMockGraphqlClient({
      Query: () => ({
        publicationCount: () => {
          return 1
        },
        reviewCount: () => {
          return 10
        },
      }),
    })
  }

  const testHarness = Vue.extend({
    components: {
      PublicationsPage,
    },
    render(h) {
      return h(PublicationsPage, { props: this.$attrs })
    },
  })

  beforeAll(() => {
    Vue.use(Vuex)
    sync(store, router)
    graphqlWithData()
  })

  it('should match snapshot', async() => {
    const wrapper = shallowMount(testHarness, { store, router, apolloProvider })
    await Vue.nextTick()

    expect(wrapper).toMatchSnapshot()
  })
})
