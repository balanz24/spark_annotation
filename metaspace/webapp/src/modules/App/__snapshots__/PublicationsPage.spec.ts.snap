// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicationsPage should match snapshot 1`] = `
<div class="sm-publications-page">
  <div class="sm-content-page sm-main-content">
    <h1>Relevant Publications</h1>
    <h2 id="ours">Our publications presenting METASPACE or its methods (5)</h2>
    <ul>
      <li>
        <h3 class="text-blue-800 text-base font-medium m-0">Spatial Metabolomics and Imaging Mass Spectrometry in the Age of Artificial Intelligence</h3>
        <p>Alexandrov T</p>
        <p class="text-sm font-medium"><span class="text-green-800">Annu. Rev. Biomed. Data Sci.</span>, <time>2020</time></p>
        <p class="text-sm truncate text-primary"><a href="https://doi.org/10.1146/annurev-biodatasci-011420-031537">https://doi.org/10.1146/annurev-biodatasci-011420-031537</a></p>
      </li>
      <li>
        <h3 class="text-blue-800 text-base font-medium m-0">OffsampleAI: artificial intelligence approach to recognize off-sample mass spectrometry images</h3>
        <p>Ovchinnikova K, Kovalev V, Stuart L, Alexandrov T</p>
        <p class="text-sm font-medium"><span class="text-green-800">BMC Bioinformatics</span>, <time>2020</time></p>
        <p class="text-sm truncate text-primary"><a href="https://doi.org/10.1186/s12859-020-3425-x">https://doi.org/10.1186/s12859-020-3425-x</a></p>
      </li>
      <li>
        <h3 class="text-blue-800 text-base font-medium m-0">ColocML: Machine learning quantifies co-localization between mass spectrometry images</h3>
        <p>Ovchinnikova K, Stuart L, Rakhlin A, Nikolenko S, Alexandrov T</p>
        <p class="text-sm font-medium"><span class="text-green-800">Bioinformatics</span>, <time>2020</time></p>
        <p class="text-sm truncate text-primary"><a href="https://doi.org/10.1093/bioinformatics/btaa085">https://doi.org/10.1093/bioinformatics/btaa085</a></p>
      </li>
      <li>
        <h3 class="text-blue-800 text-base font-medium m-0">METASPACE: A community-populated knowledge base of spatial metabolomes in health and disease</h3>
        <p>Alexandrov T, Ovchnnikova K, Palmer A, Kovalev V, et al.</p>
        <p class="text-sm font-medium"><span class="text-green-800">BioRxiv</span>, <time>2019</time></p>
        <p class="text-sm truncate text-primary"><a href="https://www.biorxiv.org/content/10.1101/539478v1.abstract">https://www.biorxiv.org/content/10.1101/539478v1.abstract</a></p>
      </li>
      <li>
        <h3 class="text-blue-800 text-base font-medium m-0">FDR-controlled metabolite annotation for high-resolution imaging mass spectrometry</h3>
        <p>Palmer A, Phapale P, Chernyavsky I, Lavigne R, Fay D, Tarasov A, Kovalev V, Fuchser J, Nikolenko S, Pineau C, Becker M, Alexandrov T</p>
        <p class="text-sm font-medium"><span class="text-green-800">Nat. Methods</span>, <time>2017</time></p>
        <p class="text-sm truncate text-primary"><a href="https://doi.org/10.1038/nmeth.4072">https://doi.org/10.1038/nmeth.4072</a></p>
      </li>
    </ul>
    <h2 id="others">Other publications</h2><a href="https://scholar.google.com/scholar?q=%22METASPACE%22+AND+%28%22imaging+mass+spectrometry%22+OR+%22mass+spectrometry+imaging%22%29" target="_blank" class="publication-link flex items-center">Research articles using METASPACE (76)<anonymous-stub class="h-4 w-4 pointer-events-none">
        <anonymous-stub></anonymous-stub>
      </anonymous-stub></a><a href="https://scholar.google.com/scholar?q=%22METASPACE%22+AND+%28%22imaging+mass+spectrometry%22+OR+%22mass+spectrometry+imaging%22%29++AND+%22Review+article%22&amp;btnG=" target="_blank" class="publication-link flex items-center mt-1">Reviews (42)<anonymous-stub class="h-4 w-4 pointer-events-none">
        <anonymous-stub></anonymous-stub>
      </anonymous-stub></a>
    <div class="cite-meta-wrapper">
      <h3>How should I cite METASPACE?</h3>
      <p class="cite-meta-p">Please cite our publication (<a href="http://www.nature.com/nmeth/journal/v14/n1/full/nmeth.4072.html">Palmer et al., 2016, Nature Methods</a>) and refer to <a href="https://metaspace2020.eu">https://metaspace2020.eu</a>.</p>
      <p>If you are preparing a scientific publication based on METASPACE annotations, create a<vuecomponent-stub to="/projects" tag="a" event="click">project</vuecomponent-stub> and follow the ‘Scientific Publishing’ workflow.</p>
    </div>
  </div>
</div>
`;
